bin/sozo/src/ops/auth.rs

```diff
@@ -16,12 +20,22 @@ pub async fn execute(command: AuthCommand, env_metadata: Option<Environment>) ->
 
                 let account = account.account(&provider, env_metadata.as_ref()).await?;
                 let world = WorldContract::new(world_address, &account);
+                let world_reader = WorldContractReader::new(world_address, &provider)
+                    .with_block(BlockId::Tag(BlockTag::Pending));
 
                 let mut calls = Vec::new();
 
                 for mc in models_contracts {
-                    let contract = get_contract_address(&world, mc.contract).await?;
-                    calls.push(world.grant_writer_getcall(&mc.model, &contract.into()));
+                    let model_name = parse_cairo_short_string(&mc.model)?;
+                    let model = world_reader.model_reader(&model_name).await?;
```

Comment:
I think creating a new model reader for an unknown/unregistered model should ideally return error but currently it doesn't.

The `model_address == FieldElement::ZERO` below should instead be the determining factor of whether creating a new model reader should return an error or not:

https://github.com/dojoengine/dojo/blob/88bae9f436acbf0325de27be68264aff979dce40/crates/dojo-world/src/contracts/model.rs#L82-L86

Reply:
Yes, good idea !
Is it better to raise an error in the world Cairo contract directly or just handle the null address in `model.rs` from `dojo-world` ?

Reply:
At the end I did the modification in `dojo-world` 👍

Reply:
> At the end I did the modification in `dojo-world` 👍

Yup good move, it makes sense to have it in the `dojo-world` to ensure the model reader is initialized on something that is actually valid.